### PR TITLE
[specific ci=1-38-Docker-Exec] Call Reset outside of the Do function

### LINKED ISSUE #5038: Integration test failures: exec echo -i and exec echo -t
From build https://ci.vmware.run/vmware/vic/10346:

```
==============================================================================
Test-Cases.Group1-Docker-Commands.1-38-Docker-Exec :: Test 1-38 - Docker Exec 
==============================================================================
Exec Echo -i                                                          | FAIL |
Server error from portlayer: Get http://127.0.0.1:2377/interaction/a989fb9c956bdc25145cbdc5e986a6020df3f12f30c9407199b7b2eb474dcd15/stderr: net/http: timeout awaiting response headers != Your eyes can deceive you. Don't trust them.
------------------------------------------------------------------------------
Exec Echo -t                                                          | FAIL |
Server error from portlayer: Get http://127.0.0.1:2377/interaction/33f77447deeade32f03d463d0cf4bc3116b7b32064fd9554550a1f2749ae380f/stdout?deadline=2017-05-05T18%3A23%3A18.673Z: net/http: timeout awaiting response headers != Do. Or do not. There is no try.
------------------------------------------------------------------------------
```

Logs:

[Test-Cases.Group1-Docker-Commands.1-38-Docker-Exec-VCH-10346-3729-container-logs.zip](https://github.com/vmware/vic/files/980176/Test-Cases.Group1-Docker-Commands.1-38-Docker-Exec-VCH-10346-3729-container-logs.zip)

### PR DESCRIPTION
**Pull request by caglar10ur**
Otherwise we get a deadlock as at that time Do holds the lock and Reset tries to get it.

Fixes #5038

### CODE REVIEW

File: lib/portlayer/attach/communication/interactor.go [hunk 1]
```diff
@@ -232,9 +235,13 @@ func (t *interaction) Unblock() error {
 		ok, err := t.channel.SendRequest(msgs.UnblockReq, true, []byte(msgs.UnblockMsg))
 		if !ok || err != nil {
 			log.Errorf("failed to unblock the other side: %s", err)
-			t.unblocked.Reset()
+			// #5038: resync package is not reentrant so we need to call Reset after this
+			atomic.StoreUint32(&reset, 1)
 		}
 	})
 
+	if atomic.LoadUint32(&reset) == 1 {
```
Comment (fdawg4l) [inline lib/portlayer/attach/communication/interactor.go]: This smells like CAS without the S.  What unsets `reset`?

Comment (fdawg4l) [inline lib/portlayer/attach/communication/interactor.go]: OH!  Nevermind.  this is a scoped variable

Comment (hickeng) [inline lib/portlayer/attach/communication/interactor.go]: may be being foolish - but do we need the atomic parts given reset is not a shared variable and there's no go routine involved. Is there memory ordering uncertainty that I'm unaware of?

Comment (caglar10ur) [inline lib/portlayer/attach/communication/interactor.go]: yeah now that you pointed out this could be a non-atmomic var., refreshed the pr

**Pull request merged**